scipy/stats/_discrete_distns.py

```diff
@@ -514,6 +514,112 @@ def _logcdf(self, k, M, n, N):
+    samples with :math:`k+r`'th sample being a failure. The former
+    can be modelled as a hypergeometric distribution. Refer
+    `hypergeom` distribution for more information.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The symbols used to denote the shape parameters (`N`, `K`, and `r`) are not
+    universally accepted. See the Examples for a clarification of the
+    definitions used here.
+
+    The probability mass function is defined as,
+
+    .. math:: f(k; N, K, r) = \frac{{{k+r-1}\choose{k}}{{N-r-k}\choose{K-k}}}
+                                   {{N \choose K}}
+
+    %(after_notes)s
+
+    Examples
+    --------
+    >>> from scipy.stats import nhypergeom
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
+
+    Suppose we have a collection of 20 animals, of which 7 are dogs.
+    Then if we want to know the probability of finding a given number
+    of dogs (successes) in a sample with exactly 12 animals that
+    aren't dogs (failures), we can initialize a frozen distribution
+    and plot the probability mass function:
+
+    >>> N, K, r = [20, 7, 12]
+    >>> rv = nhypergeom(N, K, r)
+    >>> x = np.arange(0, K)
+    >>> pmf_dogs = rv.pmf(x)
+
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111)
+    >>> ax.plot(x, pmf_dogs, 'bo')
+    >>> ax.vlines(x, 0, pmf_dogs, lw=2)
+    >>> ax.set_xlabel('# of dogs in our group with given 12 failures')
+    >>> ax.set_ylabel('nhypergeom PMF')
+    >>> plt.show()
+
+    Instead of using a frozen distribution we can also use `nhypergeom`
+    methods directly.  To for example obtain the probability mass
+    function, use:
+
+    >>> prb = nhypergeom.pmf(x, N, K, r)
+
+    And to generate random numbers:
+
+    >>> R = nhypergeom.rvs(N, K, r, size=10)
+
+    See Also
+    --------
+    hypergeom
+
+    """
+    def _get_support(self, N, K, r):
+        return 0, K
+
+    def _argcheck(self, N, K, r):
+        cond = (N >= 0)
+        cond &= (K >= 0) & (K <= N)
+        cond &= (r >= 0) & (r <= N-K)
+        return cond
+
+    def _logpmf(self, k, N, K, r):
+        result = (-betaln(k+1, r) + betaln(k+r, 1) -
+                  betaln(K-k+1, N-r-K+1) + betaln(N-r-k+1, 1) +
+                  betaln(K+1, N-K+1) - betaln(N+1, 1))
+        return result
+
+    def _pmf(self, k, N, K, r):
+        # same as the following but numerically more precise
+        # return comb(k+r-1, k) * comb(N-r-k, K-k) / comb(N, K)
+        return exp(self._logpmf(k, N, K, r))
+
+    def _stats(self, N, K, r):
+        # Promote the datatype to at least float
+        # mu = rK / (N-K+1)
+        N, K, r = 1.*N, 1.*K, 1.*r
+        mu = r*K / (N-K+1)
+
+        var = r*(N+1)*K / ((N-K+1)*(N-K+2))
+        var *= 1 - r / (N-K+1)
+
+        # The skew and kurtosis are mathematically
+        # intractable so return np.nan
+        g1, g2 = np.nan, np.nan
+        return mu, var, g1, g2
+
+
+nhypergeom = nhypergeom_gen(name='nhypergeom')
```

Comment:
From the guide:
> Does the distribution have infinite support? If not, left and/or right endpoints a, b need to be specified in the call to squirrel_gen(name='squirrel', a=?, b=?)

Probably not needed because `get_support` is implemented?

Reply:
Yes. `get_support` takes care of that.